examples/asset/multi_asset_sync.rs

```diff
@@ -0,0 +1,296 @@
+                        sections: vec![TextSection {
+                            value: "Loading...".to_owned(),
+                            style: TextStyle {
+                                font_size: 64.0,
+                                color: Color::BLACK,
+                                ..Default::default()
+                            },
+                        }],
+                        justify: JustifyText::Right,
+                        ..Default::default()
+                    },
+                    ..Default::default()
+                },
+                LoadingText,
+            ));
+        });
+
+    // Camera
+    commands.spawn(Camera3dBundle {
+        transform: Transform::from_xyz(10.0, 10.0, 15.0)
+            .looking_at(Vec3::new(0.0, 0.0, 0.0), Vec3::Y),
+        ..default()
+    });
+
+    // Light
+    commands.spawn(DirectionalLightBundle {
+        transform: Transform::from_rotation(Quat::from_euler(EulerRot::ZYX, 0.0, 1.0, -PI / 4.)),
+        directional_light: DirectionalLight {
+            shadows_enabled: true,
+            ..default()
+        },
+        ..default()
+    });
+
+    // Plane
+    commands.spawn((
+        PbrBundle {
+            mesh: meshes.add(Plane3d::default().mesh().size(50000.0, 50000.0)),
+            material: materials.add(Color::srgb(0.7, 0.2, 0.2)),
+            ..default()
+        },
+        Loading,
+    ));
+}
+
+// This showcases how to wait for assets synchronously.
+fn wait_on_load(
+    mut commands: Commands,
+    foxes: Res<OneHundredThings>,
+    barrier: Option<Res<AssetBarrier>>,
+    gltfs: Res<Assets<Gltf>>,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+) {
+    // If our barrier isn't ready, return early and wait another cycle
+    if barrier.map(|b| b.is_ready()) != Some(true) {
+        return;
+    };
+
+    // Change color of plane to green
+    commands.spawn((PbrBundle {
+        mesh: meshes.add(Plane3d::default().mesh().size(50000.0, 50000.0)),
+        material: materials.add(Color::srgb(0.3, 0.5, 0.3)),
+        transform: Transform::from_translation(Vec3::Z * -0.01),
+        ..default()
+    },));
+
+    // Spawn our scenes.
+    for i in 0..10 {
+        for j in 0..10 {
+            let index = i * 10 + j;
+            let position = Vec3::new(i as f32 - 5.0, 0.0, j as f32 - 5.0);
+            // All gltfs must exist because this is guarded by the `AssetBarrier`.
+            let gltf = gltfs.get(&foxes.0[index]).unwrap();
+            let scene = gltf.scenes.first().unwrap().clone();
+            commands.spawn(SceneBundle {
+                scene,
+                transform: Transform::from_translation(position),
+                ..Default::default()
+            });
+        }
+    }
+}
+
+// This showcases how to wait for assets asynchronously.
+fn get_async_loading_state(
+    state: Res<AsyncLoadingState>,
+    mut change_state: ResMut<NextState<LoadingState>>,
```

Comment:
I think this variable would be clearer if it was named "loading_state" or "next_loading_state"

Reply:
done!